src/viewport.cpp

```diff
@@ -1303,8 +1303,9 @@ void Viewport::OnMouseMoveEvent(const Point16 &pos)
 	}
 }
 
-WmMouseEvent Viewport::OnMouseButtonEvent(MouseButtons state)
+WmMouseEvent Viewport::OnMouseButtonEvent(MouseButtons state, WmMouseEventMode mode)
 {
+	if (mode != WMEM_PRESS) return WMME_NONE;  // Repeated buttons should be ignored for all actions below.
```

Comment:
Not sure "repeated buttons" is accurate. That is only the case if you don't press a second mouse button while holding the first mouse button.

Perhaps "Avoid processing multiple events for the same click?" (ie you get more events rather than more buttons).

There is also a connection to drag-detection, but that may be more confusing than clarifying.

Reply:
I ran into the explaining comment at the end of the review, so my comments may not be entirely optimal :p  

Still not convinced about "repeated button" though, perhaps "auto-repeat"?  That expresses more that it happens all by itself (for me at least). Keyboard also has auto-repeat iirc.